src/api/cow-subgraph/cow-subgraph.spec.ts

```diff
@@ -345,6 +340,29 @@ describe('Chain id update', () => {
     })
     cowSdk.updateChainId(SupportedChainId.GNOSIS_CHAIN)
     await cowSdk.cowSubgraphApi.getLastHoursVolume(24)
-    expect(fetchMock).toHaveBeenCalledWith(subgraphUrls[SupportedChainId.GNOSIS_CHAIN], fetchParameters)
+    expect(fetchMock).toHaveBeenCalledWith(prodUrls[SupportedChainId.GNOSIS_CHAIN], fetchParameters)
+  })
+})
+
+describe('Passing Options object', () => {
+  test('Valid: Handles Gnosis Chain staging env', async () => {
+    const fetchParameters = getFetchParameters(LAST_HOURS_VOLUME_QUERY, 'LastHoursVolume', { hours: 24 })
+    fetchMock.mockResponseOnce(JSON.stringify(LAST_24_HOURS_VOLUME_RESPONSE), {
+      status: 200,
+      headers,
+    })
+    await cowSdk.cowSubgraphApi.getLastHoursVolume(24, { chainId: SupportedChainId.GNOSIS_CHAIN, env: 'staging' })
+    const gcStagingUrl = getSubgraphUrl('staging')[SupportedChainId.GNOSIS_CHAIN]
+    expect(fetchMock).toHaveBeenCalledWith(gcStagingUrl, fetchParameters)
+  })
+
+  test("Valid: Switches to production mainnet if the specified env doesn't exist", async () => {
```

Comment:
In this case I'd prefer the api to throw instead of using a default value.

Reply:
will test the throw after making the change in another PR

Reply:
A helper that throws an exception when it does not find a variable will be useful.

```typescript
function getEnvVariable(nameVariable) {
    let value = process.env[nameVariable];
    if (!value) {
        throw (`missing environment variable ${nameVariable}`);
    }
    return val;
}
```

Reply:
we don't handle environment variables here but yes, if a user wants to use the subgraph in the staging env while on rinkeby/goerli we just throw an exception